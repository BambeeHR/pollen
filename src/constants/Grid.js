const Grid = Object.freeze({
  Widths: {
    NARROW: 'narrow',
    WIDE: 'wide',
    AUTO: 'auto',
    FULL_BLEED: 'full-bleed',
  },
});

export default Grid;
