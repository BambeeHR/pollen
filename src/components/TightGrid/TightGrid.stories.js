import Grid from '../../constants/Grid';
import TightGrid from './TightGrid.vue';

export default {
  title: 'TightGrid',
  components: TightGrid,
  parameters: {
    componentSubtitle: '',
  },
};

export const Gallery = () => ({
  components: { TightGrid },
  data() {
    return {
      widths: Object.values(Grid.Widths),
    };
  },
  template: `
  <div class="bg-gray-4">
    <TightGrid v-for="width in widths" :key="width" :width="width" class=" bg-white mb-8 shadow-6">
      <div v-for="n in 12" :key="n" class="bg-primary opacity-50 h-64" />
    </TightGrid>
  </div>
  `,
});

export const UsingTailwindClasses = () => ({
  components: { TightGrid },
  data() {
    return {
      widths: Object.values(Grid.Widths),
    };
  },
  template: `
  <div class="bg-gray-4">
    <TightGrid v-for="width in widths" :key="width" :width="width" class=" bg-white mb-8 shadow-6">
      <div class="bg-primary opacity-50 h-64 col-span-4" />
      <div class="bg-primary opacity-50 h-64 col-span-3" />
      <div class="bg-primary opacity-50 h-64 col-span-2" />
      <div class="bg-primary opacity-50 h-64 col-span-1" />
      <div class="bg-primary opacity-50 h-64 col-span-1" />
      <div class="bg-primary opacity-50 h-64 col-span-1" />
    </TightGrid>
  </div>
  `,
});
