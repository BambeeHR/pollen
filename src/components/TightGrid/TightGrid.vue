<template functional>
  <div
    class="tight-grid"
    :class="[`tight-grid--${props.width}`, data.class, data.staticClass]"
    v-bind="$options.getAttributes({ data, parent })"
    v-on="listeners"
  >
    <slot />
  </div>
</template>

<script>
import Grid from '../../constants/Grid';
import getAttributes from '../../util/getAttributes';

/**
 * import { TightGrid } from '@bambee/pollen';
 *
 * This creates a 12-column grid container `div` with a 16px (1rem) column gap.
 * Ideally this will be used with the Tailwind grid helper classes
 * (https://tailwindcss.com/docs/grid-column) but vanilla CSS for grids can also
 * be used.
 */
export default {
  getAttributes,
  props: {
    /**
     * One of `narrow`, `wide`, `auto`, and `full-bleed`. Narrow will give a
     * 768px interior width container with 16px x-axis padding. `wide` does the
     * same but with 960px interior width. `auto` fills to the container, but
     * with 16px padding, while `full-bleed` fills the container without
     * padding.
     */
    width: {
      type: String,
      default: Grid.Widths.AUTO,
      validator: (value) => Object.values(Grid.Widths).includes(value),
    },
  },
};
</script>

<style scoped>
.tight-grid {
  @apply box-border gap-4 grid grid-cols-12 mx-auto w-full;
}

.tight-grid--narrow {
  @apply px-4;
  max-width: 50rem;
}

.tight-grid--wide {
  @apply px-4;
  max-width: 62rem;
}

.tight-grid--auto {
  @apply px-4;
}
</style>
