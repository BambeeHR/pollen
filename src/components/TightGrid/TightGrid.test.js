import { mount } from '@vue/test-utils';
import Grid from '../../constants/Grid';
import TightGrid from './TightGrid.vue';

describe('TightGrid', () => {
  Object.values(Grid.Widths).forEach((width) => {
    test(`renders ${width} width correctly`, () => {
      const wrapper = mount(TightGrid, {
        propsData: { width },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
