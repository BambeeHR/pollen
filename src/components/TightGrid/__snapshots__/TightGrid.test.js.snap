// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TightGrid renders auto width correctly 1`] = `
<div
  class="tight-grid tight-grid--auto"
/>
`;

exports[`TightGrid renders full-bleed width correctly 1`] = `
<div
  class="tight-grid tight-grid--full-bleed"
/>
`;

exports[`TightGrid renders narrow width correctly 1`] = `
<div
  class="tight-grid tight-grid--narrow"
/>
`;

exports[`TightGrid renders wide width correctly 1`] = `
<div
  class="tight-grid tight-grid--wide"
/>
`;
